File: prev_exams_and_ans/2020ccc/q4_cyclic_shifts/CyclicShifts.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;

public class CyclicShifts {
    
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        
        String t = br.readLine();
        String s = br.readLine();

        // int t_len = t.length();
        // int s_len = s.length();

        ArrayList<String> s_cyclic = genCyclic(s);
        

        // WHY ??
        // for (int i = 0; i < t_len - s_len; i++) {
        //     if (s_cyclic.contains(t.substring(i, i + s_len))) {
        //         System.out.println("yes");
        //         return ;
        //     }
        // }

        for (String cyclic : s_cyclic) {
            if (t.indexOf(cyclic) >= 0) {
                System.out.println("yes");
                return ;
            }
        }

        System.out.println("no");
        return ;
    }
    
    public static ArrayList<String> genCyclic(String s) {
        int s_len = s.length();
        
        // String[] s_cyclic = new String[s_len];

        ArrayList<String> s_cyclic = new ArrayList<String>();

        for (int i = 0; i < s_len; i++) {
            s_cyclic.add(s.substring(i) + s.substring(0, i));
        }
        
        return s_cyclic;
    }
}

File: prev_exams_and_ans/2020ccc/q4_cyclic_shifts/cyclic_shifts.py
#import sys
#input = sys.stdin.readline

t = input()
s = input()

def gen_cyclic(word):
    s_len = len(word)

    s_cyclic = []
    
    for i in range(0, s_len):
        s_cyclic.append(word[i:] + word[:i])

    return s_cyclic


flag = False
for cyclic in gen_cyclic(s):
    if cyclic in t: # if t.find(cyclic) >= 0:
        print("yes")
        flag = True
        break

if (flag == False):
    print("no")

